src/unreleasedComponents/SecretTextarea.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copy & paste from https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/SecretTextArea/SecretTextArea.tsx
+ * Available starting from @grafana/ui 9.2
+ */
+
+import { css, cx } from "@emotion/css";
+import * as React from "react";
+
+import { GrafanaTheme2 } from "@grafana/data";
+import { useStyles2, Button, TextArea, HorizontalGroup } from "@grafana/ui";
```

Comment:
Which branch of grafana/grafana was this component copied from? Given grafana/ui isn't bundled in this plugin-ui any changes to the APIs of these nested components between 8 and 9 (and potentially 10) could have an impact on this component at runtime in Grafana.

Reply:
It's about the components / dependencies in ui and data that this component relies on. If none of those changed between 8, 9 and 10 it won't be an issue.

Reply:
I haven't checked all the changes that were made since v8, but I've checked it both on v8 and v9 and didn't find issues.

Reply:
>I haven't checked all the changes that were made since v8, but I've checked it both on v8 and v9 and didn't find issues.

Ok, cool! If you've tested it across Grafana >=8.4.7 without issues there shouldn't be a problem right now. I'm aware that what I'm raising here is beyond the scope of this PR and I don't think it should block it. I'll continue the conversation though to raise awareness of the pitfalls.

>All our plugins needs to be built from 8.4.7

The `yarn.lock` file in this project doesn't match this statement. It references `@grafana/<name_of_package>@8.5.9` which means the compiled NPM package code is building against 8.5.9. Because of this the [SqlDatasource](https://github.com/grafana/plugin-ui/blob/main/src/datasource/SqlDatasource.ts#L222) is using the `templateSrv.containsTemplate` method that was only introduced in 8.5.

> but needed latest components from Grafana

As time goes on this makes me think we're going to have huge portions of grafana/ui in this project depending on which nested components change.

>As this problem continues, should we move some of these composite components out of grafana core leaving only the core components such as input, select, radio and themes? Having composite components in a single place will make our life easier.

I'm not sure how this solves the problem. At runtime you're still reliant on those `input`, `select`, `theme` for a particular version of `@grafana/ui`. If any breaking changes are introduced they ripple down anyhow. 🤷